libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -272,4 +381,44 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
 
     return lowerCaseFilters;
   }
+
+  private validateRequirementType(requirement_type: any, featureFlagName: string) {
+    if(!Object.values(RequirementType).includes(requirement_type)){
+      throw new ArgumentError(
+        `This feature flag '${featureFlagName}' must have any/all requirement type.`
+      );
+    }
+
+    return true;
+  }
+
+  private getMsFmSchemaSection(config: object): any {
+    const msFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[3];
+    const featureManagementKey =  msFeatureManagementKeyWord as keyof object;
+    const featureManagementSection = config[featureManagementKey];
+
+    if (featureManagementSection && typeof featureManagementSection === "object") {
+      if (!Array.isArray(featureManagementSection[Constants.FeatureFlagsKeyWord])) {
+        throw new ArgumentError(`The ${Constants.FeatureFlagsKeyWord} key within ${msFeatureManagementKeyWord} must be an array.`);
+      }
+
+      return featureManagementSection[Constants.FeatureFlagsKeyWord];
+    }
+  }
+
+  private getLegacySchema(config: object): any {
+    const msFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[3];
+    const featureManagementSection = config[this.legacySchemaFeatureManagementKeyWord as keyof object];
+    if (featureManagementSection && typeof featureManagementSection === "object") {
```

Comment:
Looks like `typeof featureManagementSection !== "object"` is not a valid case, should we throw?

Reply:
I think we should its not valid. Updated to throw